torctl

```diff
@@ -98,6 +99,28 @@ check_root() {
     fi
 }
 
+check_tor() {
+if tor -h &> /dev/null ; then
+        sleep 0.1
+else info "Tor is not installed on your system. Installation started."
+     if pacman -Sy tor --noconfirm &> /dev/null ; then
+        sleep 0.1
+     else err "Tor installation failed."
+     fi
+fi
+}
+
+check_virtual_machine() {
+	if [ "$(dmidecode -s system-manufacturer)" == "innotek GmbH" ] &>/dev/null; then
+			warn "Virtual Machine detected" && sleep 0.1
+			if nmcli dev | grep wifi &>/dev/null; then
+					sleep 0.1
+			else
+				err "Wireless card not detected so you cannot use mac channger."  && exit 1
+			fi
```

Comment:
Not everyone uses NetworkManager, use `lshw -quiet -C network|grep "description: Wireless interface"` instead.